omnioperator/omniop-openlookeng-extension/src/main/java/nova/hetu/olk/operator/HashAggregationOmniOperator.java

```diff
@@ -316,5 +321,26 @@ public OperatorFactory duplicate()
                     groupByInputTypes, aggregationInputChannels, aggregationInputTypes, aggregatorTypes, maskChannels,
                     aggregationOutputTypes, step);
         }
+
+        @Override
+        public void checkDataType(Type type)
+        {
+            TypeSignature signature = type.getTypeSignature();
+            String base = signature.getBase();
+
+            switch (base) {
+                case StandardTypes.INTEGER:
```

Comment:
Actually when input type is RowType/Varbinary and Step is partial we need to fallback since we dont support them from data source. But when the Step is final we should not fallback since decimal intermediate states and average states are stored in these types. So the logic should be:
if (this.step == PARTIAL) {
 ...
 } else {
 ...
case ROW:
case VARBINARY:
...
default:
 fallback
}

Reply:
done